### PR TITLE
Moved Private Link Endpoints to a dedicated subnet

### PR DESCRIPTION
**Pull request by teilmeier**
- Spoke VNet Bicep template
- Container Registry Bicep template
- AKS stamp ARM template
- Network topology markdown file

### CODE REVIEW

File: networking/spoke-BU0001A0008.bicep [hunk 1]
```diff
@@ -215,6 +215,69 @@ resource nsgAppGwSubnet_diagnosticsSettings 'Microsoft.Insights/diagnosticSettin
...
+      {
+        categoryGroup: 'allLogs'
+        enabled: true
+      }
+    ]
+  }
+}
+
 // The spoke virtual network.
 // 65,536 (-reserved) IPs available to the workload, split across two subnets for AKS and one for App Gateway.
```
Comment (ckittel) [inline networking/spoke-BU0001A0008.bicep:282]: This comment will need to be updated as well

File: networking/spoke-BU0001A0008.bicep [hunk 2]
```diff
@@ -266,6 +329,17 @@ resource vnetSpoke 'Microsoft.Network/virtualNetworks@2021-05-01' = {
...
       }
+      {
+        name: 'snet-privatelinkendpoints'
+        properties: {
+          addressPrefix: '10.240.4.32/28'
+          networkSecurityGroup: {
+            id: nsgPrivateLinkEndpointsSubnet.id
+          }
+          privateEndpointNetworkPolicies: 'Disabled'
+          privateLinkServiceNetworkPolicies: 'Enabled'
```
Comment (ckittel) [inline networking/spoke-BU0001A0008.bicep:340]: I can't leave a comment on this line, but above in `snet-clusternodes`, this leaves `privateLinkServiceNetworkPolicies: 'Enabled'` -- we'll probably want to set that back to `privateLinkServiceNetworkPolicies: 'Disabled'` now that we don't have any private link endpoints showing up there.

File: networking/topology.md [hunk 3]
```diff
@@ -42,7 +44,7 @@ In the future, this VNet might hold more subnets like [ACI Provider instance] su
 
 1. [AKS System Nodepool] and [AKS User Nodepool] subnet:  Multi-tenant or other advanced workloads may have nodepool isolation requirements that might demand more (and likely smaller) subnets.
 2. [AKS Internal Load Balancer subnet]: Multi-tenant, multiple SSL termination rules, single PPE supporting dev/QA/UAT, etc could lead to needing more ingress controllers, but for baseline, we should start with one.
-3. [Private Endpoints] Private Links are created for ACR and Azure KeyVault, so these Azure services can be accessed using Private Endpoints within the Spoke vNet, specifically allocating an private Ip address from the AKS System and User Nodepool Subnet.
+3. [Private Endpoints] Private Links are created for ACR and Azure Key Vault, so these Azure services can be accessed using Private Endpoints within the Spoke vNet. There are multiple [Private Link deployment options]. In the baseline they are deployed to a dedicated subnet within the Spoke vNet.
```
Comment (ckittel) [inline networking/topology.md:47]:
```suggestion
3. [Private Endpoints] subnet: Private Links are created for Azure Container Registry and Azure Key Vault, so these Azure services can be accessed using Private Endpoints within the spoke virtual network. There are multiple [Private Link deployment options]; this this implementation they are deployed to a dedicated subnet within the spoke virtual network.
```

**Review by ckittel — CHANGES_REQUESTED** [review]
Hey there @teilmeier -- Nice work!  Just a couple small suggestions.  I like the renaming you did around the private endpoints, and keeping them consistent between both services.  Kudos.

Remind me, do you need me to try to dig up the original art?

File: networking/spoke-BU0001A0008.bicep (hunk 2 above)
Comment (teilmeier) [reply networking/spoke-BU0001A0008.bicep:341]: Updated the comment

File: networking/topology.md (hunk 3 above)
Comment (ckittel) [reply networking/topology.md]: Good catch on my typo as well :)

**Review by ckittel — APPROVED** [review]
LGTM, thanks!  When we sync today on the docs changes (and image update), we can probably push this live.  I'll hold to merge until after our chat.

**Comment by ckittel** [comment]
Docs change merged.  MicrosoftDocs/architecture-center-pr#5509

**Pull request merged**